### PR TITLE
Reverted friendly units' immunity to EMP.

### PR DESCRIPTION
**Pull request by GraionDilach**
Also removed the explicit enemy vehicle targeting.

I acknowledge, I introduced this, but tbh, I consider these features and not bugs.

### CODE REVIEW

**Comment by penev92** [comment]
Didn't EMP in TS affect all units regardless of player stance?
Also it seems pretty accurate to not care about it.

**Comment by GraionDilach** [comment]
I'm unsure, I never really used them in TS. The main reason why I filed the PR back at the time was that my mobile EMPs frequently EMPd themselves at the AI and it pretty much broke stuff.

**Comment by pchote** [comment]
You should be able to set a flag on the warhead to prevent it from affecting the actor that fired it.

**Comment by GraionDilach** [comment]
Yes, one can. Only since #8558 tho, which added this issue in the first case.

File: mods/ts/weapons/superweapons.yaml [hunk 1]
```diff
@@ -90,7 +90,7 @@ EMPulseCannon:
 		Range: 3c0
 		Duration: 250
 		Upgrades: empdisable
-		ValidTargets: Vehicle
+		ValidStances: Ally, Neutral, Enemy
```
Comment (abcdefg30) [inline mods/ts/weapons/superweapons.yaml]: You don't need to explicitly set this, as `ValidStances` has this values as defaults.

Comment (GraionDilach) [inline mods/ts/weapons/superweapons.yaml]: Oh, right. For some reason I remembered having only Neutral and Enemy.

**Comment by GraionDilach** [comment]
Removed unnecessary line.

**Comment by abcdefg30** [comment]
Now EMPs target also buildings etc., but that is your intention? So :+1: / :white_check_mark:

**Comment by GraionDilach** [comment]
For now, yes. We don't have a separate TargetType defined for terrain alone (Ground also lets building to be targeted).

It was also the status before #8558.

**Comment by obrakmann** [comment]
:+1:

**Pull request merged**